config.toml

```diff
@@ -2,25 +2,47 @@
 
 # (Required) Core settings apply to the service as a whole rather than to individual components within it.
 [core_settings]
-# Size of the thread pool used for processing requests. Defaults to the number of processors on 
+# Size of the thread pool used for processing requests. Defaults to the number of processors on
 # the machine.
 #thread_pool_size = 8
 
 # Duration of sleep when the connection pool is empty. This can limit the response
 # times for requests and so should be set to a low number. Default value is 10.
 #idle_listener_sleep_duration = 10 # in milliseconds
 
+# Log level to be applied across the service. Can be overwritten for certain modules which have the same
+# configuration key. Possible values: "debug", "info", "warn", "error", "trace"
+#log_level = "warn"
+
+# Control whether log entries contain a timestamp.
+#log_timestamp = false
+
 # (Required) Configuration for the service IPC listener component.
 [listener]
-# (Required) Defines the type of IPC that the service will support. 
+# (Required) Type of IPC that the service will support.
 listener_type = "DomainSocket"
-# (Required) Defines the timeout of the read and write operations on the IPC channel. After the
+
+# (Required) Timeout of the read and write operations on the IPC channel. After the
 # timeout expires, the connection is dropped.
 timeout = 200 # in milliseconds
 
-# (Required) Configuration for the component managing key IDs for providers.
-[key_manager]
-# (Required) Defines the type of key ID manager to be used.
+# (Required) Configuration for the components managing key IDs for providers.
+# Defined as an array of tables: https://github.com/toml-lang/toml#user-content-array-of-tables
+[[key_manager]]
```

Comment:
Was thinking about that. The thing is that despite this being an array, the contents of the array are added as: 

```
[[key_manager]]
...first key manager...

[[key_manager]]
...second key manager...
```

So in a way it looks more sensible to keep the name as singular, since each "header" identifies only one entry.